README.md

```diff
@@ -1,234 +1,278 @@
+- **config/** - Configurações do projeto
+- **cube/** - Projeto do STM32CubeMX (.ioc e arquivos gerados)
+- **include/** - Cabeçalhos
+- **docs/** - Documentação gerada (não versionado)
+- **lib/** - Submódulos e bibliotecas externas
+- **src/** - Código fonte principal da aplicação
+- **test/** - Testes
+
+## 🛠 Configuração
+
+### 1. Projeto CubeMX
+
+1. Crie um novo projeto na pasta `cube/`
+2. Configurações:
+    - **Project > Application Structure:** Basic
+    - **Project > Toolchain/IDE:** CMake
+    - **Code Generator > Generate peripheral initialization:** Pair of .c/.h
+    - **Code Generator > Delete previous generated files:** Ativado
+
+### 2. CMakeLists.txt
+
+Edite o arquivo principal `CMakeLists.txt` com as informações do seu projeto:
+
+```cmake
+# Nome do projeto (igual ao arquivo .ioc sem extensão)
+set(CMAKE_PROJECT_NAME meu_projeto)
+
+# Versão da placa (opcional)
+set(BOARD_VERSION "")
+```
 
-* uncrustify
-  > Linux: `sudo apt install uncrustify`
-  >
-  > Windows: Baixe o .zip no [SourceForge](https://sourceforge.net/projects/uncrustify/files/). Adicione o local do executável na variável de ambiente `PATH`.
+## 🔨 Compilação
 
-* [Visual Studio Code](https://code.visualstudio.com/)
-  * [EditorConfig](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)
-  * [C/C++](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
-  * [Cortex-Debug](https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug)
+Antes de iniciar, crie uma pasta `build/` na raiz do projeto
 
-* [STM32 Cube Programmer](https://www.st.com/en/development-tools/stm32cubeprog.html) ou [J-Link](https://www.segger.com/downloads/jlink/)
-  > É necessário que o executável também esteja no `PATH`
+```bash
+mkdir build
+cd build
+```
 
-## Preparando
+Dentro dela, configure o ambiente com
 
-### Projeto
+```bash
+cmake ..
+```
 
-Primeiro é necessário criar um projeto do Cube na pasta `cube/` com o nome desejado,
-que deve ter as seguintes opções de projeto:
+Depois, compile o projeto
 
-Project:
+```bash
+make -j
+```
 
-* Application Structure: *Basic*
-* [x] Do not generate the main()
-* Toolchain / IDE: *Makefile*
+> O parâmetro `-j` ativa a compilação paralela, usando mais núcleos do seu processador
 
-Code Generator:
+### Limpar arquivos
 
-* STM32Cube Firmware Library Package: *Copy all used libraries into the project folder*
-* Generated files:
-  * *Generate peripheral initialization as a pair of .c/.h files per peripheral*
-  * *Delete previously generated files when not re-generated*
+```bash
+make clear       # Código do usuário
+make clear_cube  # Bibliotecas Cube
+make clear_all   # Tudo
+```
 
-Um arquivo de exemplo se encontra em `cube/stm32_project_template.ioc` com todas as configurações necessárias.
+### Manual
 
-Para projetos existentes, basta mover o arquivo `.ioc` para a pasta `cube/`.
+Para obter uma lista completa de comandos, use
 
-### Compilação
+```bash
+make help
```

Comment:
```suggestion
make helpme
```

Nao sei se foi proposital, mas make help printa só uma lista de todos os targets, o make helpme printa o textinho bonitinho que ta no cmake/templates/helpme.in

Reply:
Nossa, isso n faz sentido nenhum, não é possível que não dê pra desabilitar o default do cmake